src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/modals/transactionRuleModal.tsx

```diff
@@ -36,26 +35,13 @@ type State = Form['state'] & {
 
 class TransactionRuleModal extends Form<Props, State> {
   componentDidUpdate(prevProps: Props, prevState: State) {
-    if (prevState.tracing !== this.state.tracing && !!this.state.conditions.length) {
```

Comment:
As the UI now disables the tracing option when there are rules, this condition is no longer needed.